recipes/libgit2/0.27.x/conanfile.py

```diff
@@ -38,43 +41,42 @@ class LibGit2Conan(ConanFile):
         "with_sha1": "collisiondetection",
     }
 
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake", "cmake_find_package"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-        if not tools.is_apple_os(self.settings.os):
+        if not is_apple_os(self):
             del self.options.with_iconv
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     @property
     def _need_openssl(self):
         return "openssl" in (self.options.with_https, self.options.with_sha1)
 
     def requirements(self):
-        self.requires("zlib/1.2.12")
+        self.requires("zlib/1.2.13")
         self.requires("http_parser/2.9.4")
         if self.options.with_libssh2:
-            self.requires("libssh2/1.10.0")
+            self.requires("libssh2/1.11.0")
         if self.settings.os != "Windows":
-            self.requires("libcurl/7.83.1")
+            self.requires("libcurl/8.1.2")
```

Comment:
```suggestion
            self.requires("libcurl/[>=X.YY <9]")
```

https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges